starter-code/Bookshelf/app/src/main/java/ly/generalassemb/drewmahrt/bookshelf/MainActivity.java

```diff
@@ -36,27 +41,61 @@ protected void onCreate(Bundle savedInstanceState) {
         mBookAdapter = new BaseAdapter() {
             @Override
             public int getCount() {
+
                 return mBookList.size();
             }
 
             @Override
             public Object getItem(int position) {
+
                 return mBookList.get(position);
             }
 
             @Override
             public long getItemId(int position) {
+
                 return position;
             }
 
             @Override
             public View getView(int position, View convertView, ViewGroup parent) {
                 //TODO: Update the view
+
+                if (counterView == null) {
+                    LayoutInflater inflater = LayoutInflater.from(MainActivity.this);
+                    convertView = inflater.inflate(android.R.layout.simple_list_item_2, null);
+
+                }
+
+                TextView textView = (TextView) convertView.findViewById(android.R.id.text1);
+                textView.setText(mBookList.get(position).getTitle());
+
+                TextView textView2 = (TextView) counterView.findViewById(android.R.id.text2);
+                textView2.setText(mBookList.get(position).getAuthor());
+
+                return convertView;
             }
         };
 
         //TODO: Set the ListView's adapter
 
+        listView.setAdapter(mBookList);
+        mBookAdapter.notifyDataSetChanged();
+
+        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                TextView textView = (TextView) view.findViewById(android.R.id.text1);
+                TextView textView2 = (TextView) view.findViewById(android.R.id.text2);
+                textView.setTextColor(Color.RED);
+                textView2.setTextColor(Color.RED);
+                mBookAdapter.notifyDataSetChanged();
+            }
+                                        }
```

Comment:
I think we have an issue with mis-matched brackets here - somewhere you should have `});` in order to conclude the statement that began with `listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {`